source/binary.cpp

```diff
@@ -541,7 +546,9 @@ spv_result_t Parser::parseOperand(size_t inst_offset,
                 setNumericTypeInfoForType(&parsed_operand, inst->type_id))
           return error;
       }
-      break;
+      if (auto error = verifyLiteralUpperBits(parsed_operand,word))
```

Comment:
This should be done in the validator instead of in the binary parser.

Why?   The binary parser is the lowest layer: both the disassembler and the validator are built on top of it.  The disassembler is used to help debug problems.  In a bad case you're catching, it's usually preferable to see a disassembly rather than a truncated failed parse.

So, the binary parser generally does as little as possible.  It only does enough to ensure well-formedness, with validity being a whole other bunch of work.

Since you can defer these checks to the validator, you should do so.

And I realize none of this is written down where you'd be able to find it.   :-(

Reply:
I targeted the binary parser because there things happen at word level.
As a result, I could handled all literals with a same function, a same logic.

The validator works at instruction level, with different logics checking each of them.
As a result, now I need to inject new code to every instruction accepting literals.

right @dneto0? or do we have a common place in the validator where to check literals?